src/services/expenses.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+let expenses = [];
+let id = 0;
+
+const getAll = ({ userId, category, from, to }) => {
+  if (!expenses.length) {
+    return [];
+  }
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === +userId);
+  }
+
+  if (category) {
+    expenses = expenses.filter(expense => expense.category === category);
+  }
+
+  if (from) {
+    expenses = expenses.filter(expense => expense.spentAt >= from);
+  }
+
+  if (to) {
+    expenses = expenses.filter(expense => expense.spentAt <= to);
+  }
```

Comment:
You make 4 iterations to filter expenses, figure out how to decrease them to 1

Reply:
Ok